meson.build

```diff
@@ -439,6 +440,10 @@ foreach decl : ['char16_t',
         conf.set10('HAVE_' + decl.underscorify().to_upper(), have)
 endforeach
 
+conf.set10('HAVE_STRUCT_STATX_IN_SYS_STAT_H', cc.sizeof('struct statx', prefix : '''
+#include <sys/stat.h>
+''', args : '-D_GNU_SOURCE') > 0)
```

Comment:
hmm, i am pretty sure we should set _GNU_SOURCE for all out tests, no, not just this one?

@keszybz ?

Reply:
Yeah, we should.

Reply:
https://github.com/systemd/systemd/pull/9643